cypress/support/commands.js

```diff
@@ -394,8 +394,14 @@ Cypress.Commands.add('createUser', user => {
 	user.roles.forEach(role => {
 		cy.get('form[id=userRoleForm]').contains(role).click();
 	});
-	cy.get('form[id=userRoleForm] button[id^=submitFormButton]').click();
-	cy.get('span[id$="-username"]:contains("' + Cypress.$.escapeSelector(user.username) + '")');
+	cy.server();
+	cy.route('POST', '**/grid/settings/user/user-grid/update-user-roles*').as('rolesSaved');
+	cy.get('div[id^=component-grid-settings-user-usergrid]').as('userGrid').then(staleUserGrid => {
+		cy.get('form[id=userRoleForm] button[id^=submitFormButton]').click();
+		cy.wait('@rolesSaved').its('status').should('eq', 200);
+		// Wait for the interface to refresh, the "User Grid" will be recreated and get a new ID.
+		cy.get('@userGrid').should('not.have.attr', 'id', staleUserGrid.attr('id'));
```

Comment:
Here it's basically just waiting for the list component to get re-rendered, the purpose is to avoid a `cy.wait()`.
I got curious to search for alternative fixes, probably something using Vue's nextTick()/life-cycle might be good enough for our use-case.

Reply:
Because this is a grid, it should be possible to use the `waitJQuery()` function (see `lib/pkp/cypress/support/commands.js`). But you could also argue as a UX pattern that a newly-created record *should* appear on the page after it's created, i.e. perhaps the grid should jump to the last page or similar.

Reply:
Yeah, I was going to try the `waitJQuery()`, but I thought it was a Vue interface and end up not trying :)

In fact I was mostly trying to find a generic way to solve flaky tests, the `cy.get('selector').should('be.visible')` is supposed to retry the `cy.get()`, but it didn't, even a `cy.get('BODY selector').should('be.visible')` to ensure the element is attached to the DOM failed 😪